Sofa/Component/Constraint/Lagrangian/Correction/src/sofa/component/constraint/lagrangian/correction/LinearSolverConstraintCorrection.inl

```diff
@@ -575,16 +576,36 @@ template<class DataTypes>
 void LinearSolverConstraintCorrection<DataTypes>::addConstraintDisplacement(double *d, int begin, int end)
 {
     const MatrixDeriv& constraints = mstate->read(core::ConstMatrixDerivId::constraintJacobian())->getValue();
-    const auto derivDim = Deriv::size();
 
     last_disp = begin;
 
-
-    l_linearSolver.get()->partial_solve(Vec_I_list_dof[last_disp], Vec_I_list_dof[last_force], _new_force);
+    l_linearSolver->partial_solve(Vec_I_list_dof[last_disp], Vec_I_list_dof[last_force], _new_force);
 
     _new_force = false;
 
-    const auto positionIntegrationFactor = l_ODESolver.get()->getPositionIntegrationFactor();
+    // Lambda function adding the constraint displacement using [] if a FullVector is detected or element() else
+    constexpr auto addConstraintDisplacement_impl = [](double* d, unsigned int id, auto* systemLHVector_buf, double positionIntegrationFactor, unsigned int dof, const Deriv& val)
```

Comment:
Interesting use of constexpr auto lambda but Isn't it much more clear and common to use a template function ?